specs/core/0_beacon-chain.md

```diff
@@ -729,205 +723,191 @@ def int_sqrt(n: int) -> int:
         sig=proof_of_possession,
         domain=get_domain(
             fork_data,
             current_slot,
             DOMAIN_DEPOSIT
         )
     )
-    new_validators = copy.deepcopy(current_validators)
-    validator_pubkeys = [v.pubkey for v in new_validators]
-
-    # add new validator
+    validators_copy = copy.deepcopy(validators)
+    validator_pubkeys = [v.pubkey for v in validators_copy]
+    
     if pubkey not in validator_pubkeys:
-        assert deposit_size == DEPOSIT_SIZE
-
-        rec = ValidatorRecord(
+        # Add new validator
+        validator = ValidatorRecord(
             pubkey=pubkey,
             withdrawal_credentials=withdrawal_credentials,
             randao_commitment=randao_commitment,
             randao_skips=0,
-            balance=DEPOSIT_SIZE * GWEI_PER_ETH,
+            balance=deposit,
             status=status,
             last_status_change_slot=current_slot,
-            exit_seq=0
+            exit_counter=0
         )
 
-        index = min_empty_validator(new_validators)
+        index = min_empty_validator_index(validators_copy)
         if index is None:
-            new_validators.append(rec)
-            index = len(new_validators) - 1
+            validators_copy.append(validator)
+            index = len(validators_copy) - 1
         else:
-            new_validators[index] = rec
-        return new_validators, index
-
-    # topup existing validator
+            validators_copy[index] = validator
     else:
+        # Increase balance by deposit
         index = validator_pubkeys.index(pubkey)
-        val = new_validators[index]
-        assert deposit_size >= MIN_TOPUP_SIZE
-        assert val.status != WITHDRAWN
-        assert val.withdrawal_credentials == withdrawal_credentials
+        validator = validators_copy[index]
+        assert validator.withdrawal_credentials == withdrawal_credentials
+
+        validator.balance += deposit
 
-        val.balance += deposit_size
-        return new_validators, index
+    return validators_copy, index
 ```
 
 Now, to add a validator or top up an existing validator's balance:
 
 ```python
-def add_or_topup_validator(state: BeaconState,
-                           pubkey: int,
-                           deposit_size: int,
-                           proof_of_possession: bytes,
-                           withdrawal_credentials: Hash32,
-                           randao_commitment: Hash32,
-                           status: int,
-                           current_slot: int) -> int:
+def process_deposit(state: BeaconState,
+                    pubkey: int,
+                    deposit: int,
+                    proof_of_possession: bytes,
+                    withdrawal_credentials: Hash32,
+                    randao_commitment: Hash32,
+                    status: int,
+                    current_slot: int) -> int:
     """
     Add the validator into the given `state`.
```

Comment:
This function was `add_validator`, but now it also accepts top-up so this line of docstring should be changed or removed.

Reply:
Thanks 👍